### DISCUSSION TITLE
Webpage crashes when scrollZoom is enabled for Map

### DISCUSSION THREAD

**Discussion by zeel91297** [uber/react-map-gl#1386] [Q&A]
Hi, My page is crashed every time I try to zoom in or zoom out. If I disable `scrollZoom` it is working fine.
The thing is I want to implement it with scrolling also.

`"mapbox-gl": "^1.13.1",
"react-map-gl": "^5.3.10",`

<img width="1069" alt="Screen Shot 2021-03-02 at 2 38 41 PM" src="https://user-images.githubusercontent.com/28985505/109625156-fd353200-7b64-11eb-90a5-a8ba991a41c0.png">

Can anyone please help me with this?

**Comment by Pessimistress** [comment]
We cannot debug from a screenshot. Please create a CodeSandbox that reproduces your issue.

**Comment by zeel91297** [comment]
@Pessimistress Here is the link to CodeSandbox
https://codesandbox.io/s/funny-field-9lo5d?fontsize=14&hidenavigation=1&theme=dark

If you're not able to see the above error then please keep open networks tab and try to zoom-in and zoom-out.

**Comment by Pessimistress** [comment]
Try not to mutate the incoming viewport. I.e. instead of

`const newViewPort = nextViewport;`

Use

`const newViewPort = {...nextViewport};`

Before you overwrite its content.

**Comment by zeel91297** [comment]
Thanks a lot!
